src/main/java/seedu/rc4hdb/model/resident/fields/House.java

```diff
@@ -0,0 +1,75 @@
+package seedu.rc4hdb.model.resident.fields;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.rc4hdb.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Resident's house in RC4HDB.
+ * Guarantees: immutable; is valid as declared in {@link #isValidHouse(String)}
+ */
+public class House {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "House should only contain the following characters: 'D', 'U', 'L', 'A' or 'N'";
+
+    public static final String VALIDATION_REGEX = "^[DULAN]$";
+
+    public final String house;
+
+    /**
+     * Constructs a {@code House}.
+     *
+     * @param house A valid house.
+     */
+    public House(String house) {
+        requireNonNull(house);
+        checkArgument(isValidHouse(house), MESSAGE_CONSTRAINTS);
+        this.house = house;
+    }
+
+    /**
+     * Returns true if a given string is a valid house.
+     */
+    public static boolean isValidHouse(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns the house string as a full word.
+     * @return the house string as a full word.
+     */
+    public String asFullWord() {
+        switch (house) {
+        case ("D"):
```

Comment:
Some formatting issues here. Also, should we omit the parentheses for the case clauses? I believe `case "D":` would suffice.

Reply:
The same comments apply to the other cases for the rest of the switch statement.